### PR TITLE
Remove deprecated classes and methods in quantum_info

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Remove
- deprecated `operators.Pauli`, `pauli_group`
- deprecated methods in `Pauli`

### CODE REVIEW

**Comment by ikkoham** [comment]
The CI error has already been resolved by https://github.com/Qiskit/qiskit-ignis/pull/571.
The tests use stable, so failed. What should I do? (Should I wait the release of Ignis?)

**Comment by mtreinish** [comment]
> The CI error has already been resolved by [Qiskit/qiskit-ignis#571](https://github.com/Qiskit/qiskit-ignis/pull/571).
> The tests use stable, so failed. What should I do? (Should I wait the release of Ignis?)

Yeah, we'll have to wait for a release of ignis. The tutorials CI job is doing it's job and catching things that would break users if they were going to upgrade to terra with this pr. If we merged this today and potentially released before ignis, anyone using the cnot dihedral rb from pypi would be broken. The ignis release is blocked on #6867 though because we're going to deprecate ignis in it's next release.

**Comment by ikkoham** [comment]
Thank you. Let's wait.

**Comment by jakelishman** [comment]
Are we scheduling this for Terra 0.20, since Ignis 0.7 (deprecation boogaloo) can't release until Terra 0.19?

**Comment by ikkoham** [comment]
If we release Ignis 0.7 first and then Terra 0.19, can't we include this PR in the next releases? (there are many other reasons, so if that's not possible, 0.20 is fine.)

**Comment by mtreinish** [comment]
It's unfortunately not possible to sequence things that way because the release of ignis 0.7 is blocked on the terra 0.19.0 release.

**Comment by ikkoham** [comment]
Thank you. This PR should be merged for 0.20.

**Comment by ikkoham** [comment]
A part of this PR is splitted into #7291.

File: qiskit/quantum_info/operators/dihedral/dihedral.py [hunk 1]
```diff
@@ -314,6 +313,8 @@ def to_circuit(self):
                *Scalable randomised benchmarking of non-Clifford gates*,
                npj Quantum Inf 2, 16012 (2016).
         """
+        from qiskit.quantum_info.synthesis.cnotdihedral_decompose import decompose_cnotdihedral
```
Comment (Cryoris) [inline qiskit/quantum_info/operators/dihedral/dihedral.py:316]: Does this import need to be local due to some new cyclic dependency?

**Comment by Cryoris** [comment]
Since Ignis 0.7 was released in December, can we proceed with this PR? 🙂 

@ikkoham Could you add a releasenote stating that this deprecated code is now removed? I think this is usually part of the `upgrade` section 🙂

**Comment by mtreinish** [comment]
I'm going to close this since it's been superseded by  #8070 . If this is incorrect please feel free to reopen this.

**Pull request closed without merging**